### PR TITLE
Release 3.5.5 release notes

### PR DESCRIPTION
**Pull request by awholegunch**
Release notes for release 3.5.5. 

(Note: the large number of changes in the file(s) are because there were a large number of lines with trailing whitespace, which has been removed to match our style guide.)

Running as a Quick Build: https://builds.cask.co/browse/CDAP-DQB323-4

Page of interest: 
- **Release Notes:** https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/reference-manual/release-notes.html#release-3-5-5
- **Security Properties:** https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/admin-manual/appendices/cdap-site.html#security
- **Deprecated Properties:** https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/admin-manual/appendices/cdap-site.html#deprecated-properties
- **Security Docs:** _Enabling Authorization:_ https://builds.cask.co/artifact/CDAP-DQB323/shared/build-4/Docs-HTML/3.5.5/en/admin-manual/security/authorization.html#enabling-authorization

### CODE REVIEW

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -30,6 +30,50 @@ Cask Data Application Platform Release Notes
...
+=============================================================
+
+New Features
+------------
+- :cask-issue:`CDAP-4535` - Authentication server announce address is now configurable
+  with the property ``security.auth.server.announce.urls``, which are comma-separated URLs in
+  the form of ``protocol://host:port``. The property
+  ``security.auth.server.announce.address`` is now deprecated. It is only used if it is set
+  but ``security.auth.server.announce.urls`` has not been set.
+  ``security.auth.server.announce.address`` takes a single address in the form of
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: `security.auth.server.announce.address` this property should already be documented right and work as expected? no need to document it again in the release notes.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -30,6 +30,50 @@ Cask Data Application Platform Release Notes
...
+  ``host:port`` or ``host``. Protocol and port (if not specified) will be added by the
+  Authentication Server to form the URL. A default URL will be generated by the
+  Authentication Serve if either property is not set.
+
+- :cask-issue:`CDAP-8990` - New configurations have been added to control the YARN
+  application master container memory size, maximum heap memory size, and maximum non-heap
+  memory size: ``twill.java.heap.memory.ratio``, ``twill.yarn.am.memory.mb``, and
+  ``twill.yarn.am.reserved.memory.mb``.
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst:52]: nit: the parameters should be in order of the description
1. application master container memory size->`twill.yarn.am.memory.mb`
2. maximum heap memory size->`twill.java.heap.memory.ratio`
3. maximum non-heap memory size->`twill.yarn.am.reserved.memory.mb`

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -30,6 +30,50 @@ Cask Data Application Platform Release Notes
...
+- :cask-issue:`CDAP-9005` - Fixed an issue where ``HBaseQueueDebugger`` failed when
+  authorization was enabled.
+
+- :cask-issue:`CDAP-9160` - Fixed a memory leak issue with the Hadoop FileSystem object.
+
+- :cask-issue:`CDAP-9085` - Fixed an out-of-memory issue for the log saver by adding a
+  limit on the maximum number of events in-memory.
+
+- :cask-issue:`CDAP-8997` - Fixed an issue with uncaught exceptions so that they are
+  logged through the logger, allowing log collections for those exce
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: nit: `for those exceptions.`

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/release-notes.rst:52]: Actually, no. I had a looong talk with Terence about this. Those do not correspond, which was my first thought. I need to add docs explaining how these work.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 3 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/release-notes.rst]: Not a nit. Good catch. Error in original Release Note on confluence (trailing line break that was encoded.)

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/release-notes.rst]: No, this has not been documented in this detail before, nor elsewhere in the documentation. There is also a change in that the deprecated property (``security.auth.server.announce.address``) now accepts either `host:port` or just `host`.

File: cdap-common/src/main/resources/cdap-default.xml [hunk 4]
```diff
@@ -1667,9 +1667,10 @@
     <name>security.auth.server.announce.urls</name>
     <value></value>
     <description>
-      CDAP Authentication service announce URL's separated by comma. Each URL is in the format of protocol://host:port.
-      These are the URL's that clients should use to communicate with the Authentication Server.
-      Leave empty to use the default value generated by the Authentication Server.
+      A comma-separated list of CDAP Authentication service announce URLs. Each URL is in
+      the form of protocol://host:port. These are the URLs that clients should use to
+      communicate with the Authentication Server. Leave empty to use the default value
+      generated by the Authentication Server.
```
Comment (ecfm) [inline cdap-common/src/main/resources/cdap-default.xml]: Maybe also mention `If deprecated property "security.auth.server.announce.address" is set and this property is empty, "security.auth.server.announce.address" will be used`

Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml]: OK.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 5]
```diff
@@ -30,6 +30,52 @@ Cask Data Application Platform Release Notes
...
+  with the property ``security.auth.server.announce.urls``, which are comma-separated URLs in
+  the form of ``protocol://host:port``. The property
+  ``security.auth.server.announce.address`` is now deprecated. It is only used if it is set
+  but ``security.auth.server.announce.urls`` has not been set.
+
+  ``security.auth.server.announce.address`` now takes a single address in the form of either
+  ``host:port`` or ``host``. Protocol and port (if not specified) will be added by the
+  Authentication Server to form the URL.
+
+  A default URL will be generated by the Authentication Serve if either property is not set.
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: nit: `Authentication Server`

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 6]
```diff
@@ -30,6 +30,52 @@ Cask Data Application Platform Release Notes
...
+New Features
+------------
+- :cask-issue:`CDAP-4535` - Authentication server announce address is now configurable
+  with the property ``security.auth.server.announce.urls``, which are comma-separated URLs in
+  the form of ``protocol://host:port``. The property
+  ``security.auth.server.announce.address`` is now deprecated. It is only used if it is set
+  but ``security.auth.server.announce.urls`` has not been set.
+
+  ``security.auth.server.announce.address`` now takes a single address in the form of either
+  ``host:port`` or ``host``. Protocol and port (if not specified) will be added by the
```
Comment (sagarkapare) [inline cdap-docs/reference-manual/source/release-notes.rst]: Please remove the last line `Protocol and port....`

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (sagarkapare) [reply cdap-docs/reference-manual/source/release-notes.rst:52]: I still think its better to have it as - 

New configurations have been added to control the YARN application master container memory size, maximum heap memory size, and maximum non-heap memory size: ``twill.yarn.am.memory.mb``, ``twill.java.heap.memory.ratio``, and ``twill.yarn.am.reserved.memory.mb``.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 5 above)
Comment (awholegunch) [reply cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

**Pull request merged**